azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -1761,6 +1761,20 @@ private void ajaxHandleUpload(final HttpServletRequest req, final HttpServletRes
       throws ServletException, IOException {
     final User user = session.getUser();
     final String projectName = (String) multipart.get("project");
+
+    // Fetch the uploader's IP
+    String uploaderIPAddr = "";
+    if (req != null) {
+      uploaderIPAddr = req.getHeader("X-FORWARDED-FOR");
+      if (uploaderIPAddr == null || uploaderIPAddr.isEmpty()) {
+        logger.debug("Failed to fetch remote Address using \"X-FORWARDED-FOR\"");
+        uploaderIPAddr = req.getRemoteAddr();
```

Comment:
Why we're getting the IP address from the request object and not from the header itself?